flybywiresim/a32nx
# feat(vnav): initial check-in of VNAV, only for DECEL pseudo waypoint support

## Summary of Changes

This merges in the WIP VNAV codebase in order to support `(DECEL)` in cFMS v1.

Most of the VNAV functionality is disabled via `VnavConfig.ts`, as follows:

```ts
export const VnavConfig = {

    /**
     * Whether to calculate climb profile
     */
    VNAV_CALCULATE_CLIMB_PROFILE: false,

    /**
     * Whether to emit ToD pseudo waypoint
     */
    VNAV_EMIT_TOD: false,

    /**
     * Whether to emit (DECEL) pseudo waypoint
     */
    VNAV_EMIT_DECEL: true,

    /**
     * VNAV descent calculation mode (NORMAL, CDA or DPO)
     */
    VNAV_DESCENT_MODE: VnavDescentMode.NORMAL,

    /**
     * Whether to emit CDA flap1/2 pseudo-waypoints (only if VNAV_DESCENT_MODE is CDA)
     */
    VNAV_EMIT_CDA_FLAP_PWP: false,

};
```

## Additional context

Discord username (if different from GitHub): someperson#4953

## Testing instructions

* Make sure `(DECEL)` pseudo-waypoint appears on ND and MCDU F-PLN page
* Make sure `(T/D)`, `(FLAP1)` and `(FLAP2)` pseudo-waypoints **do not** appear on ND and MCDU F-PLN page
* Make sure APPROACH **phase** (not mode!) auto-engages when passing `(DECEL)` when in DES **phase** (not mode!)


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
